sections/dashboard/Overview/Overview.tsx

```diff
@@ -24,62 +20,56 @@ import PortfolioChart from '../PortfolioChart';
 import SpotMarketsTable from '../SpotMarketsTable';
 import SynthBalancesTable from '../SynthBalancesTable';
 
+export enum PositionsTab {
+	CROSS_MARGIN = 'cross margin',
+	ISOLATED_MARGIN = 'isolated margin',
+	SPOT = 'spot',
+}
+
 const Overview: FC = () => {
 	const { t } = useTranslation();
 
-	const { useSynthsBalancesQuery } = useSynthetixQueries();
-
-	const portfolioValueQuery = useGetCurrentPortfolioValue();
-	const portfolioValue = portfolioValueQuery?.data ?? null;
-
-	const futuresPositionQuery = useGetFuturesPositionForAccount();
-	const futuresPositionHistory = futuresPositionQuery?.data ?? [];
-
-	const exchangeRates = useRecoilValue(ratesState);
-
-	const { walletAddress } = Connector.useContainer();
-	const synthsBalancesQuery = useSynthsBalancesQuery(walletAddress);
-	const synthBalances =
-		synthsBalancesQuery.isSuccess && synthsBalancesQuery.data != null
-			? synthsBalancesQuery.data
-			: null;
+	const balances = useRecoilValue(balancesState);
+	const portfolio = useRecoilValue(portfolioState);
+	const positions = useRecoilValue(positionsState);
 
-	const [activePositionsTab, setActivePositionsTab] = useRecoilState(activePositionsTabState);
+	const [activePositionsTab, setActivePositionsTab] = useRecoilState<PositionsTab>(
+		activePositionsTabState
+	);
 	const [activeMarketsTab, setActiveMarketsTab] = useState<MarketsTab>(MarketsTab.FUTURES);
 
-	const totalSpotBalancesValue = formatDollars(wei(synthBalances?.totalUSDBalance ?? zeroBN));
-
-	const totalFuturesPortfolioValue = formatDollars(wei(portfolioValue ?? zeroBN));
-
-	const POSITIONS_TABS = useMemo(
-		() => [
+	const POSITIONS_TABS = useMemo(() => {
+		const crossPositions = positions.cross_margin.filter(({ position }) => !!position).length;
```

Comment:
Do you think it would be better to filter out null positions at the source or we need to also show market positions with empty positions in places?

Reply:
We need to keep those positions to total the remaining margin. This handles situations where you have margin but no open position. They are included so we can total margin, but not count them as a position on the badge.